openff/qcsubmit/_tests/results/test_results.py

```diff
@@ -379,13 +379,9 @@ def test_optimization_to_basic_result_collection(public_client):
     optimization_result_collection = OptimizationResultCollection.from_server(
         public_client, ["OpenFF Gen 2 Opt Set 3 Pfizer Discrepancy"]
     )
-    with (
-        TemporaryDirectory() as d,
-        portal_client_manager(lambda a: _CachedPortalClient(a, d)),
-    ):
-        basic_collection = optimization_result_collection.to_basic_result_collection(
-            "hessian"
-        )
+    basic_collection = optimization_result_collection.to_basic_result_collection(
+        "hessian"
+    )
```

Comment:
(not blocking)

> I think it might be a good idea to modify at least one of the tests using portal_client_manager to ensure that a plain PortalClient works. I changed every test to use _CachedPortalClient when it was going to become the new default in https://github.com/openforcefield/openff-qcsubmit/pull/286 to avoid sharing the same cache directories, but now it should be safe to get rid of some of those TemporaryDirectory and portal_client_manager calls again.

Yeah, it's good that you caught that `default_portal_client` was going to go uncovered and changed some of these back. It seems a little brittle that a newcomer wouldn't understand why some tests are using `portal_client_manager` and others aren't. But that can be a topic for a future PR (maybe a dedicated test for default/non-default clients? Or some pattern of using `pytest.parameterize` that doesn't double runtime?). If you can think of a clean comment to leave explaining this (like `# this test doesn't use portal_client_manager just so it can test out the default pathway`) could you add it before merging?

Reply:
Should I include this comment in each of the four tests that no longer use `portal_client_manager`? I guess I was thinking that since that is the default behavior, it makes sense for it to be the default test behavior too. Possibly I should instead comment on the one test still using `portal_client_manager` (`test_to_records`) to point out why it's being used there?

Maybe it would have been better if I had just copy-pasted `test_to_records` to begin with to have a dedicated `test_cached_to_records` test as you mention. I thought I would only have to make small changes to `test_to_records` at first, but in the end it basically became a different test anyway.

Reply:
Let's put a comment on the one test that uses caching, like "as of Aug 2024, this is the one place we're testing caching"